src/sentry/sentry_metrics/indexer/postgres/postgres_v2.py

```diff
@@ -195,8 +261,13 @@ def reverse_resolve(self, use_case_id: UseCaseKey, org_id: int, id: int) -> Opti
         string: str = obj.string
         return string
 
-    def _table(self, use_case_id: UseCaseKey) -> IndexerTable:
-        return TABLE_MAPPING[use_case_id]
+    def _get_metric_path_key(self, use_case_ids: Collection[UseCaseID]) -> UseCaseKey:
+        if UseCaseID.SESSIONS in use_case_ids:
+            return UseCaseKey.RELEASE_HEALTH
+        return UseCaseKey.PERFORMANCE
+
+    def _get_table(self, use_case_ids: Collection[UseCaseID]) -> IndexerTable:
+        return TABLE_MAPPING[self._get_metric_path_key(use_case_ids)]
```

Comment:
i'm a little concerned by this logic, can we at least assert that `len(use_case_ids) == 1` (or have more sophisticated logic around returning a collection of tables from a collection of UseCaseIDs)

Reply:
For more sophisticated logic, what do you have in mind?

Reply:
a `Mapping[UseCaseID,IndexerTable]` would probably be the multi-use-case-supporting return type but I don't think we actually want to support that

Really you just want to make sure that all your input `UseCaseID` values map to a single `UseCaseKey` value. so let's do that assertion somewhere (either that `len(use_case_ids)==1` or `len(set(self._get_metric_path_key(id) for id in use_case_ids)) == 1`

Reply:
> a Mapping[UseCaseID,IndexerTable] would probably be the multi-use-case-supporting return type but I don't think we actually want to support that
My (very superficial) understanding is that we will never need to write to both table in a single `bulk_record` call, so a `Mapping[UseCaseID,IndexerTable]` would never be the multi-use-case-supporting return type, because multi-use-case would still be writing to a single table (performance). Maybe @ayirr7 can correct me

Reply:
I think your understanding is right, but it does support my case that we should fail hard if a collection of use_case_ids passed here map to two different `IndexerTable`s

Reply:
Sounds good I agree, I added a check that throws a `ValueError` because it's more descriptive that way